File: apps/commerce-api/src/main/java/com/loopers/infrastructure/usersignal/UserSignalCoreRepository.java
package com.loopers.infrastructure.usersignal;

import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Component;

import com.loopers.domain.usersignal.TargetType;
import com.loopers.domain.usersignal.UserSignal;
import com.loopers.domain.usersignal.UserSignalRepository;

import lombok.RequiredArgsConstructor;

@Component
@RequiredArgsConstructor
public class UserSignalCoreRepository implements UserSignalRepository {
    private final UserSignalJpaRepository userSignalJpaRepository;

    @Override
    public UserSignal save(UserSignal userSignal) {
        return userSignalJpaRepository.save(userSignal);
    }

    @Override
    public Optional<UserSignal> find(TargetType type, Long targetId) {
        return userSignalJpaRepository.findByTargetForUpdate(type, targetId);
    }

    @Override
    public List<UserSignal> findAllIn(List<Long> productIds, TargetType targetType) {
        return userSignalJpaRepository.findAllByTargetIdIn(productIds, targetType);
    }
}

File: apps/commerce-streamer/src/main/kotlin/com/loopers/infrastructure/usersignal/UserSignalCoreRepository.kt
package com.loopers.infrastructure.usersignal

import com.loopers.domain.usersignal.TargetType
import com.loopers.domain.usersignal.UserSignal
import com.loopers.domain.usersignal.UserSignalRepository
import org.springframework.stereotype.Component

@Component
class UserSignalCoreRepository(
    val userSignalJpaRepository: UserSignalJpaRepository
) : UserSignalRepository {

    override fun findForUpdate(targetId: Long, targetType: TargetType): UserSignal? {
        return userSignalJpaRepository.findForUpdate(targetId, targetType)
    }
}
